Libraries/LibCrypto/BigInt/UnsignedBigInteger.cpp

```diff
@@ -64,18 +64,71 @@ UnsignedBigInteger UnsignedBigInteger::add(const UnsignedBigInteger& other)
     return result;
 }
 
+UnsignedBigInteger UnsignedBigInteger::sub(const UnsignedBigInteger& other)
+{
+    UnsignedBigInteger result;
+
+    if (*this < other) {
+        dbg() << "WARNING: bigint subtraction creates a negative number!";
+        return UnsignedBigInteger::create_invalid();
+    }
+
+    u8 borrow = 0;
+    for (size_t i = 0; i < other.length(); ++i) {
+        ASSERT(!(borrow == 1 && m_words[i] == 0));
+
+        if (m_words[i] - borrow < other.m_words[i]) {
+            u64 after_borrow = static_cast<u64>(m_words[i] - borrow) + (UINT32_MAX + 1);
+            result.m_words.append(static_cast<u32>(after_borrow - static_cast<u64>(other.m_words[i])));
+            borrow = 1;
+        } else {
+            result.m_words.append(m_words[i] - borrow - other.m_words[i]);
+            borrow = 0;
+        }
+    }
+
+    for (size_t i = other.length(); i < length(); ++i) {
+        ASSERT(!(borrow == 1 && m_words[i] == 0));
```

Comment:
@itamar8910, This assert trips when
```
UnsignedBigInteger { 1 }.shift_left(256).sub(1)
```

Reply:
Are you sure you're using the latest commit? I fixed a bug in operator< that also tripped this assert in 9b90722

Reply:
Does iterating `other.length()` times make sense here? it might have leading zeros.

`a.divide(b)` trips a vector OOB assert, it could be because of this
```
b: 0|51657| a: 65535|
```